octodns/provider/yaml.py

```diff
@@ -102,7 +107,97 @@ def _apply(self, plan):
         if not isdir(self.directory):
             makedirs(self.directory)
 
+        self._do_apply(desired, data)
+
+    def _do_apply(self, desired, data):
         filename = join(self.directory, '{}yaml'.format(desired.name))
         self.log.debug('_apply:   writing filename=%s', filename)
         with open(filename, 'w') as fh:
             safe_dump(dict(data), fh)
+
+
+# Any record name added to this set will be included in the catch-all file,
+# instead of a file matching the record name.
+_CATCHALL_RECORD_NAMES = ('*', '')
+
+
+def list_all_yaml_files(directory):
+    yaml_files = set()
+    for f in listdir(directory):
+        filename = join(directory, '{}'.format(f))
+        if f.endswith('.yaml') and isfile(filename):
+            yaml_files.add(filename)
+    return list(yaml_files)
+
+
+class SplitYamlProvider(YamlProvider):
+    '''
+    Core provider for records configured in multiple YAML files on disk.
+
+    Behaves mostly similarly to YamlConfig, but interacts with multiple YAML
+    files, instead of a single monolitic one. The files are named RECORD.yaml,
+    except for any record which cannot be represented easily as a file; these
+    are stored in the catchall file, which is a YAML file the zone name,
+    prepended with a '$'. For example, a zone, 'github.com.' would have a
+    catch-all file named '$github.com.yaml'.
+
+    config:
+        class: octodns.provider.yaml.SplitYamlProvider
+        # The location of yaml config files (required)
+        directory: ./config
+        # The ttl to use for records when not specified in the data
+        # (optional, default 3600)
+        default_ttl: 3600
+        # Whether or not to enforce sorting order on the yaml config
+        # (optional, default True)
+        enforce_order: True
+    '''
+
+    def __init__(self, id, directory, *args, **kwargs):
+        super(SplitYamlProvider, self).__init__(id, directory, *args, **kwargs)
+        self.log = logging.getLogger('SplitYamlProvider[{}]'.format(id))
```

Comment:
I believe the `self.log =` needs to come before the super call so that the parent class can use it.

Reply:
That's a little tricky, because if specified _before_ calling `super(..)__init__(..)`, the logger gets blown away with the new one. I've modified the logger creation in `YamlProvider` to be dynamic, and removed it from `SplitYamlProvider` entirely. Let me know what you think.

Reply:
👍 that's the right way to go.